attestation-service/verifier/src/cca/mod.rs

```diff
@@ -80,13 +119,13 @@ impl Verifier for CCA {
         expected_init_data_hash: &InitDataHash,
     ) -> Result<TeeEvidenceParsedClaim> {
         let ReportData::Value(expected_report_data) = expected_report_data else {
-            bail!("CCA verifier must provide report data field!");
+            return Err(CCAError::ReportDataMissing.into());
```

Comment:
I notice that severval `.into()` are used in this function to convert `Result<_, CCAError>` to `Result<_, anyhow::Error>`.

One way to avoid this is to have an inner non-pub function `cca_evaluate() -> Result<_, CCAError>` to be wrapped by this `evaluate()`. In `cca_evaluate()` can delete all the `.into()`s. And in `evaluate()` just call `cca_evaluate()` to finish the error conversion.

Another way is to add another overall error type `VerifierError` under `verifier`. One enum is named `CCA`. And change the signature of `evaluate(..)->anyhow::Result<TeeEvidenceParsedClaim>` to `evaluate(..)->Result<TeeEvidenceParsedClaim, VerifierError>`. In this function different `CCAError`s will finally be mapped into `VerifierError::CCA`.

It's up to you : )